From 7473f90007f6e88b85421ad42ae2b4e77f104683 Mon Sep 17 00:00:00 2001
From: Christian Ehrhardt <christian.ehrhardt@canonical.com>
Date: Thu, 1 Jun 2017 11:32:39 +0200
Subject: [PATCH 2/2] Show Distro specific SMTP banner.

Patch to add the Distribution name to the default SMTP banner.
That allows e.g. market researchers to detect the distribution running a
server more easily (nobody wants to be in the "other" bucket).

Signed-off-by: Christian Ehrhardt <christian.ehrhardt@canonical.com>
---
 debian/patches/fix_smtp_banner.patch | 52 ++++++++++++++++++++++++++++++++++++
 debian/patches/series                |  1 +
 2 files changed, 53 insertions(+)
 create mode 100644 debian/patches/fix_smtp_banner.patch

diff --git a/debian/patches/fix_smtp_banner.patch b/debian/patches/fix_smtp_banner.patch
new file mode 100644
index 0000000..c891540
--- /dev/null
+++ b/debian/patches/fix_smtp_banner.patch
@@ -0,0 +1,52 @@
+Description: Add EXIM_DISTRIBUTION var to display it on the SMTP banner
+Origin: https://blueprints.launchpad.net/ubuntu/+spec/servercloud-s-server-app-banner-updates
+Author: Yolanda Robla <yolanda.robla@canonical.com>
+Last-Update: 2015-07-06
+
+--- a/src/globals.c
++++ b/src/globals.c
+@@ -1295,7 +1295,7 @@
+ uschar *smtp_active_hostname   = NULL;
+ BOOL    smtp_authenticated     = FALSE;
+ uschar *smtp_banner            = US"$smtp_active_hostname ESMTP "
+-                             "Exim $version_number $tod_full"
++                             "Exim $version_number " EXIM_DISTRIBUTION " $tod_full"
+                              "\0<---------------Space to patch smtp_banner->";
+ BOOL    smtp_batched_input     = FALSE;
+ BOOL    smtp_check_spool_space = TRUE;
+--- a/src/config.h.defaults
++++ b/src/config.h.defaults
+@@ -204,4 +204,6 @@
+ #define SC_EXIM_ARITH "%" SCNi64		/* scanf incl. 0x prefix */
+ #define SC_EXIM_DEC   "%" SCNd64		/* scanf decimal */
+ 
++#define EXIM_DISTRIBUTION
++
+ /* End of config.h.defaults */
+--- a/scripts/Configure-config.h
++++ b/scripts/Configure-config.h
+@@ -23,6 +23,12 @@
+ if [ "$1" != "" ] ; then MAKE=$1 ; fi
+ if [ "$MAKE" = "" ] ; then MAKE=make ; fi
+ 
++# exporting distribution to use it in smtp banner
++if test -x /usr/bin/lsb_release && lsb_release -si; then
++	export EXIM_DISTRIBUTION=\"$(lsb_release -si)\"
++else
++	export EXIM_DISTRIBUTION=\"\"
++fi
+ $MAKE buildconfig || exit 1
+ 
+ # BEWARE: tab characters needed in the following sed command. They have had
+--- a/src/exim.h
++++ b/src/exim.h
+@@ -596,5 +596,9 @@
+ # undef DISABLE_DNSSEC
+ #endif
+ 
++#ifndef EXIM_DISTRIBUTION
++  #define EXIM_DISTRIBUTION ""
++#endif
++
+ #endif
+ /* End of exim.h */
diff --git a/debian/patches/series b/debian/patches/series
index cc060c5..a794af4 100644
--- a/debian/patches/series
+++ b/debian/patches/series
@@ -9,3 +9,4 @@
 67_unnecessaryCopt.diff
 70_remove_exim-users_references.dpatch
 78_Disable-chunking-BDAT-by-default.patch
+fix_smtp_banner.patch
-- 
2.7.4

